activiti-cloud-modeling-service/activiti-cloud-services-modeling/activiti-cloud-services-modeling-rest/src/main/java/org/activiti/cloud/services/modeling/rest/controller/ModelController.java

```diff
@@ -232,6 +233,71 @@ public void validateModelExtensions(
         }
     }
 
+    @Override
+    public PagedModel<EntityModel<Model>> getGlobalModels(
+        @RequestParam(MODEL_TYPE_PARAM_NAME) String type,
+        @RequestParam(value = INCLUDE_ORPHANS_PARAM_NAME, required = false) Boolean includeOrphans,
+        Pageable pageable) {
+
+        return pagedCollectionModelAssembler.toModel(
+            pageable,
+            modelService.getGlobalModels(findModelType(type),includeOrphans,pageable),
+            representationModelAssembler);
+    }
+
+    @Override
+    public EntityModel<Model> putProjectModelRelationship(
+        @PathVariable String projectId,
+        @PathVariable String modelId,
+        @RequestParam(value = SCOPE_PARAM_NAME, required = false) String scope,
+        @RequestParam(value = FORCE_PARAM_NAME, required = false) Boolean force) {
+
+        Project project = projectController.findProjectById(projectId);
+        Model modelToBeUpdated = findModelById(modelId);
+        Model newModel = findModelById(modelId);
```

Comment:
It sounds strange to me use `findModelById(modelId)` to initialize both `modelToBeUpdated` and `newModel`. It's probably reaching the cache, but otherwise, it would mean that we ha two access to the database to retrieve the same object at the same place. I was wondering if you cannot use the same variable in the update. I mean `modelService.updateModel(newModel, newModel)`.

BTW, as the controller is annotated with `@Transactional`, I think the only reason for that you need to call `modelService.updateModel` is because it has some validation logic. Otherwise, your changes will be probably be taken into account event without calling `updateModel`.

Reply:
Regarding your 2nd concern, there are further validations in fact (https://github.com/Activiti/activiti-cloud/blob/db266925733464f6465f5465a609e4200f1b7cfe/activiti-cloud-modeling-service/activiti-cloud-services-modeling/activiti-cloud-services-modeling-service/src/main/java/org/activiti/cloud/services/modeling/service/ModelServiceImpl.java#L178)

Regarding first, you are right, in this case there is no need for having two different variables.